docs/guides/writing-tests.md

```diff
@@ -0,0 +1,409 @@
+```
+
+### Assertions
+
+If you’ve ever accused Professor Plum of murder in the library with the lead pipe while playing Clue, you’ve made an assertion. The assertion is the grand conclusion of your test, where you state what you think will happen based on the results of your query.
+
+ui-test-utils uses chai’s `expect()` assertion style (https://www.chaijs.com) to make assertions on components. In addition, the following custom assertions are available to make assertions on a query result:
+
+- `expect(await find([selector]))`
+- `.to.have.tagName([tagName])`
+- `.to.have.id([id])`
+- `.to.have.style([property], [value])`
+- `.to.have.attribute([attribute], [value]).to.have.label([label])`
+- `.to.have.title([title])`
+- `.to.have.value([value])`
+- `.to.exist()`
+- `.to.have.className([className])`
+- `.to.be.disabled()`
+- `.to.be.readonly()`
+- `.to.have.role([role])`
+- `.to.be.selected()`
+- `.to.be.checked()`
+- `.to.match([selector])`
+- `.to.be.focused()`
+- `.to.have.focus()`
+- `.to.be.focusable()`
+- `.to.be.tabbable()`
+- `.to.be.visible()`
+- `.to.be.clickable()`
+- `.to.have.text([text])`
+- `.to.have.bounds([property], [value])`
+- `.to.have.exactly([count]).descendants([selector])`
+- `.to.have.exactly([count]).ancestors([selector])`
+- `.to.contain([selector|element])`
+- `.to.be.empty()`
+
+### Testing for accessibility
+
+Once you've mounted your component in the document you can run tools like axe-core (https://www.deque.com/axe/) to verify that the rendered markup meets accessibility requirements. `ui-test-utils` provides a wrapper utility for running axe-core:
+
+```javascript
+import { accessible, mount, expect } from '@instructure/ui-test-utils'
+
+it('should be accessible', async () => {
+  await mount(<MyComponent />)
+  expect(await accessible()).to.be.true()
+})
+```
+
+### Testing responsive components
+
+Testing the behavior of responsive components often requires changing the browser viewport size. ui-test-utils provides a utility to do just that:
+
+```javascript
+import { viewport, stub } from '@instructure/ui-test-utils'
+
+it('should do something when the viewport size changes', async () => {
+  const handleResize = stub()
+  await mount(<MyComponent onResize={handleResize} />)
+
+  // sets the viewport to 320px x 480px
+  viewport.set(320, 480)
+
+  expect(handleResize).to.have.been.calledOnce()
+})
+```
+
+The test sandbox is configured to call viewport.reset() in between each test, so you don’t have to worry about resetting the viewport size.
+
+### Locators: page object models for components
+
+**A locator is a collection of query methods and CSS selectors that makes your tests easier to maintain.**
+
+If you’re familiar with the concept of Page Object Models (https://martinfowler.com/bliki/PageObject.html), it might be helpful to think of locators as POMs for UI components. Locators ideally live with their components, making it easy to keep them up-to-date when the component implementation is updated.
```

Comment:
```suggestion
If you’re familiar with the concept of [Page Object Models](https://martinfowler.com/bliki/PageObject.html), it might be helpful to think of locators as POMs for UI components. Locators ideally live with their components, making it easy to keep them up-to-date when the component implementation is updated.
```